test/test_dcrt.py

```diff
@@ -2,159 +2,267 @@
+    )
+    vi_screening = dcrt_pvalue(
+        results_screening[0],
+        results_screening[1],
+        results_screening[2],
+        results_screening[3],
     )
-    assert len(results_no_screening[1]) == 10
-    assert len(results_no_screening[2]) == 10
-    assert len(results_screening[1]) == 10
-    assert len(results_screening[2]) == 10
+    assert np.sum(results_no_screening[0]) == 10
+    assert np.sum(results_screening[0]) < 10
+    assert len(vi_no_screening[0]) <= 10
+    assert len(vi_no_screening[1]) == 10
+    assert len(vi_no_screening[2]) == 10
+    assert len(vi_screening[0]) <= 10
+    assert len(vi_screening[1]) == 10
+    assert len(vi_screening[2]) == 10
 
+    # Checking with scaled statistics
+    vi_scaled = dcrt_pvalue(
+        results_no_screening[0],
+        results_no_screening[1],
+        results_no_screening[2],
+        results_no_screening[3],
+    )
+    assert len(vi_scaled[0]) <= 10
+    assert len(vi_scaled[1]) == 10
+    assert len(vi_scaled[2]) == 10
+
+
+def test_dcrt_lasso_with_estimed_coefficient(generate_regation_dataset):
+    """
+    Test the estimated coefficient parameter
+    """
+    X, y = generate_regation_dataset
     # Checking with random estimated coefficients for the features
-    rng = np.random.RandomState(2024)
+    rng = np.random.RandomState(2025)
     estimated_coefs = rng.rand(10)
 
-    res = dcrt_zero(
+    results = dcrt_zero(
         X,
         y,
         estimated_coef=estimated_coefs,
         screening=False,
-        verbose=True,
         statistic="residual",
-        random_state=2024,
+        random_state=2026,
     )
-    assert len(res[1]) == 10
-    assert len(res[2]) == 10
+    vi = dcrt_pvalue(results[0], results[1], results[2], results[3])
+    assert len(vi[0]) <= 10
+    assert len(vi[1]) == 10
+    assert len(vi[2]) == 10
 
+
+def test_dcrt_lasso_with_refit(generate_regation_dataset):
+    """
+    Test the refit parameter
+    """
+    X, y = generate_regation_dataset
     # Checking with refit
     results_refit = dcrt_zero(
         X,
         y,
         refit=True,
-        screening=True,
-        screening_threshold=50,
-        verbose=True,
+        screening=False,
         statistic="residual",
         random_state=2024,
     )
-    assert len(results_refit[1]) == 10
-    assert len(results_refit[2]) == 10
+    vi_refit = dcrt_pvalue(
+        results_refit[0],
+        results_refit[1],
+        results_refit[2],
+        results_refit[3],
+    )
+    assert len(vi_refit[0]) <= 10
+    assert len(vi_refit[1]) == 10
+    assert len(vi_refit[2]) == 10
 
-    # Checking with scaled statistics
-    results_scaled = dcrt_zero(
+
+def test_dcrt_lasso_with_use_cv(generate_regation_dataset):
+    """
+    Test the use_cv parameter
+    """
+    X, y = generate_regation_dataset
+    # Checking with use_cv
+    results_use_cv = dcrt_zero(
         X,
         y,
-        scaled_statistics=True,
+        n_alphas=20,
```

Comment:
if this is costly, we could consider to reduce this number

Reply:
no, it's fine.